stackslib/src/chainstate/coordinator/mod.rs

```diff
@@ -845,9 +845,21 @@ pub fn get_reward_cycle_info<U: RewardSetProvider>(
                 .expect("FATAL: no start-of-prepare-phase sortition");
 
         let mut tx = sort_db.tx_begin()?;
-        if SortitionDB::get_preprocessed_reward_set(&mut tx, &first_prepare_sn.sortition_id)?
-            .is_none()
-        {
+        let preprocessed_reward_set =
+            SortitionDB::get_preprocessed_reward_set(&mut tx, &first_prepare_sn.sortition_id)?;
+        let need_to_store = if let Some(reward_cycle_info) = preprocessed_reward_set {
```

Comment:
Can you add a bit more inline comment here about what issue this change fixes and how? I think what’s happening here is that (1) the chains coordinator can first mark a reward set as unknown and then it gets stored there, and then (2) when it gets recomputed after the anchor block arrives, this check allows it to overwrite.

Does this PR have a regression test for this behavior?

Reply:
Technically, yes -- a slew of other tests would fail to run if this behavior wasn't added (namely, the downloader tests would all spin forever for want of a computed reward cycle)

Reply:
They began to stall once we removed the aggregate public key and began to instead rely on inspecting the three most recent reward sets directly in order to obtain the keys needed to authenticate downloaded blocks.  The stall was induced by the changes in the PR into which this PR is targeted.

Reply:
Okay, that makes sense.

This code block just needs some more comments then.

Reply:
Done